visualization/src/resource-visualization.cc

```diff
@@ -256,31 +306,55 @@ void visualizeReprojectedDepthResource(
           return;
         }
 
+        if (FLAGS_vis_pointcloud_filter_dense_map_before_publishing) {
+          pcl::PointCloud<pcl::PointXYZI>::Ptr pcl_cloud(
+              new pcl::PointCloud<pcl::PointXYZI>);
+          CHECK_NOTNULL(pcl_cloud);
+          backend::convertPointCloudType(points_S, &(*pcl_cloud));
+          /*
+          voxelGridPointCloud(
+              FLAGS_vis_pointcloud_filter_leaf_size_m, pcl_cloud);
+          backend::convertPointCloudType((*pcl_cloud), &points_S);
+          */
+        }
+
         // Either publish in local frame with a tf or in global frame.
         if (FLAGS_vis_pointcloud_publish_in_sensor_frame_with_tf) {
+          sensor_msgs::PointCloud2 ros_point_cloud_S;
           if (FLAGS_vis_pointcloud_color_random) {
             resources::PointCloud points_C_colorized = points_S;
             accumulated_point_cloud_G.colorizePointCloud(r, g, b);
             sensor_msgs::PointCloud2 ros_point_cloud_S;
             backend::convertPointCloudType(
                 points_C_colorized, &ros_point_cloud_S);
-            publishPointCloudInLocalFrameWithTf(T_G_S, &ros_point_cloud_S);
           } else {
-            sensor_msgs::PointCloud2 ros_point_cloud_S;
             backend::convertPointCloudType(points_S, &ros_point_cloud_S);
-            publishPointCloudInLocalFrameWithTf(T_G_S, &ros_point_cloud_S);
           }
+          if (FLAGS_vis_pointcloud_filter_dense_map_before_publishing) {
```

Comment:
Here I am confused by the flow of the if statements. From the config it's not obvious to me that `FLAGS_vis_pointcloud_filter_dense_map_before_publishing` and `FLAGS_vis_pointcloud_filter_beautify_dense_map_before_publishing` are exclusive, but they seem to be so. If they are exclusive, there should be a check and warning for this. If they are meant to work together this needs to be reworked.

Reply:
`FLAGS_vis_pointcloud_filter_dense_map_before_publishing` is deprecated as we anyway can do this with the beautification. I removed the respective parts.